vscode/src/autoedits/providers/fireworks.ts

```diff
@@ -0,0 +1,74 @@
+import { type AutoEditsTokenLimit, logDebug } from '@sourcegraph/cody-shared'
+import type * as vscode from 'vscode'
+import type {
+    AutocompleteContextSnippet,
+    DocumentContext,
+} from '../../../../lib/shared/src/completions/types'
+import type {
+    ChatPrompt,
+    PromptProvider,
+    PromptProviderResponse,
+    PromptResponseData,
+} from '../prompt-provider'
+import { getModelResponse } from '../prompt-provider'
+import { SYSTEM_PROMPT, getBaseUserPrompt } from '../prompt-utils'
+
+export class FireworksPromptProvider implements PromptProvider {
+    getPrompt(
+        docContext: DocumentContext,
+        document: vscode.TextDocument,
+        context: AutocompleteContextSnippet[],
+        tokenBudget: AutoEditsTokenLimit
+    ): PromptResponseData {
+        const { codeToReplace, promptResponse: userPrompt } = getBaseUserPrompt(
+            docContext,
+            document,
+            context,
+            tokenBudget
+        )
+        const prompt: ChatPrompt = [
+            {
+                role: 'system',
+                content: SYSTEM_PROMPT,
+            },
+            {
+                role: 'user',
+                content: userPrompt,
+            },
+        ]
+        return {
+            codeToReplace: codeToReplace,
```

Comment:
When the property name and the variable name are the same, you can just write it once as a shorthand.
```suggestion
            codeToReplace,
```

Reply:
done